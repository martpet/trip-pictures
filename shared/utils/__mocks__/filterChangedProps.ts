export const filterChangedProps = vi.fn().mockName('filterChangedProps').mockReturnValue({
  dummyChangedProp: 'dummyChangedPropValue',
});
