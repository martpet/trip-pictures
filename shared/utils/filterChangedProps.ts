export const filterChangedProps = <
  NewProps extends Record<string, any>,
  OldProps extends Record<string, any>
>(
  newProps: NewProps,
  oldProps: OldProps
) => {
  const changedProps: Partial<NewProps & OldProps> = {};

  Object.keys(newProps).forEach((key) => {
    const newVal = newProps[key];
    const oldVal = oldProps[key];

    if (newVal !== oldVal) {
      changedProps[key as keyof typeof changedProps] = newVal;
    }
  });

  if (Object.keys(changedProps).length) {
    return changedProps;
  }

  return undefined;
};
