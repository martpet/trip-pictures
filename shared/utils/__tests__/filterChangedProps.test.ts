import { filterChangedProps } from '../filterChangedProps';

describe('lowercase', () => {
  it('return a correct value', () => {
    expect(
      filterChangedProps({ a: 'a', b: 'b2', d: 'd' }, { a: 'a', b: 'b', c: 'c' })
    ).toMatchSnapshot();
  });

  it('return a correct value when no props are changed', () => {
    expect(filterChangedProps({ a: 'a' }, { a: 'a' })).toMatchSnapshot();
  });
});
