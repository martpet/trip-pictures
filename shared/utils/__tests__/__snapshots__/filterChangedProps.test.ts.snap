// Vitest Snapshot v1

exports[`lowercase > return a correct value 1`] = `
{
  "b": "b2",
  "d": "d",
}
`;

exports[`lowercase > return a correct value when no props are changed 1`] = `undefined`;
